pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -116,6 +133,12 @@ func (c *customRouteController) sync(ctx context.Context, syncCtx factory.SyncCo
 		return fmt.Errorf("custom route configuration failed verification: %v", errors)
 	}
 
+	if oidcAvailable, err := common.ExternalOIDCConfigAvailable(c.authLister, c.kasLister, c.kasConfigMapLister); err != nil {
+		return err
+	} else if oidcAvailable {
+		return c.removeOperands(ctx, ingressConfigCopy, secretName)
```

Comment:
Why do we do it that late? We don't need `secretName`. We could start immediately after the deep copy of ingressConfig, could we?

Reply:
Good catch! This is leftover from an initial version where I deleted the secret manually but then I reused the syncer.

Reply:
Fixed with 4200edde03363dde52a1473266ce0f0b22be825d